lib/liquid/filters.ex

```diff
@@ -27,57 +42,465 @@ defmodule Liquid.Filters do
+      plus value, to_number(operand)
+    end
+
+    def plus(value, operand) do
+      value |> to_number |> plus(to_number(operand))
+    end
+
+    def minus(value, operand) when is_number(value) and is_number(operand) do
+      value - operand
+    end
+
+    def minus(value, operand) when is_number(value) do
+      minus value, to_number(operand)
+    end
+
+    def minus(value, operand) do
+      value |> to_number |> minus(to_number(operand))
+    end
+
+    def times(value, operand) when is_integer(value) and is_integer(operand) do
+      value * operand
+    end
+
+    def times(value, operand) do
+      {value_int, value_len} = value |> get_int_and_counter
+      {operand_int, operand_len} = operand |> get_int_and_counter
+      case value_len + operand_len do
+        0 -> value_int * operand_int
+        precision -> (value_int * operand_int / :math.pow(10, precision)) |> Float.round(precision)
+      end
+    end
+
+    def divided_by(input, operand) when is_number(input) do
+      case {input, operand |> to_number} do
+        {_, 0} -> raise ArithmeticError, message: "divided by 0"
+        {input, number_operand} when is_integer(input) ->
+          input / number_operand |> floor
+        {input, number_operand} ->
+          input / number_operand
+      end
+    end
+
+    def divided_by(input, operand) do
+      input |> to_number |> divided_by(operand)
+    end
+
+
+    def floor(input) when is_integer(input), do: input
+
+    def floor(input) when is_number(input), do: input |> trunc
+
+    def floor(input), do: input |> to_number |> floor
+
+    def floor(input, precision) when is_number(precision) do
+      input |> to_number |> Float.floor(precision)
+    end
+
+    def floor(input, precision) do
+      input |> floor(to_number(precision))
+    end
+
+
+    def ceil(input) when is_integer(input), do: input
+
+    def ceil(input) when is_number(input) do
+      input |> Float.ceil |> trunc
+    end
+
+    def ceil(input), do: input |> to_number |> ceil
+
+    def ceil(input, precision) when is_number(precision) do
+      input |> to_number |> Float.ceil(precision)
+    end
+
+    def ceil(input, precision) do
+      input |> ceil(to_number(precision))
+    end
+
+    def round(input) when is_integer(input), do: input
+
+    def round(input) when is_number(input) do
+      input |> Float.round |> trunc
+    end
+
+    def round(input), do: input |> to_number |> round
+
+    def round(input, precision) when is_number(precision) do
+      input |> to_number |> Float.round(precision)
+    end
+
+    def round(input, precision) do
+      input |> round(to_number(precision))
+    end
+
+
+    @doc """
+    Allows you to specify a fallback in case a value doesn’t exist.
+    `default` will show its value if the left side is nil, false, or empty
+    """
+    @spec default(any, any) :: any
+    def default(input, default_val\\"")
+
+    def default(input, default_val) when is_nil(input), do: default_val
```

Comment:
You can fold all these functions into one by checking  for default in [nil, false, ""]  etc.